LeaderboardBackend/Services/LeaderboardService.cs

```diff
@@ -0,0 +1,38 @@
+using LeaderboardBackend.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace LeaderboardBackend.Services
+{
+    public class LeaderboardService : ILeaderboardService
+    {
+        private LeaderboardContext _leaderboardContext;
+        private IConfiguration _config;
+        public LeaderboardService(LeaderboardContext leaderboardContext, IConfiguration config)
+        {
+            _leaderboardContext = leaderboardContext;
+            _config = config;
+        }
+
+        public async Task<Leaderboard?> GetLeaderboard(long id)
+        {
+            Leaderboard? leaderboard = await _leaderboardContext.Leaderboards.FindAsync(id);
+            return leaderboard;
+        }
+
+        // FIXME: Paginate this
+        public async Task<List<Leaderboard>> GetLeaderboards(long[] ?ids = null)
+        {
+            if (ids == null) {
+                return await _leaderboardContext.Leaderboards.ToListAsync();
+            } else {
+                return await _leaderboardContext.Leaderboards.Where(l => ids.Contains(l.Id)).ToListAsync();
+            }
+        }
+
+        public async Task CreateLeaderboard(Leaderboard leaderboard)
+        {
+            _leaderboardContext.Leaderboards.Add(leaderboard);
```

Comment:
Not a real review comment but writing it down so it's in our heads: we should start adding validation to the models before we create them. We could call some `leaderboard.Validate()` method before creating them in this spot, would be a pretty good spot to enforce it.

Reply:
Made issue from this comment :+1: